pilot/pkg/xds/pcds.go

```diff
@@ -0,0 +1,62 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package xds
+
+import (
+	mesh "istio.io/api/mesh/v1alpha1"
+	"istio.io/istio/pilot/pkg/model"
+	tb "istio.io/istio/pilot/pkg/trustbundle"
+	"istio.io/istio/pkg/util/gogo"
+)
+
+// PcdsGenerator generates ECDS configuration.
+type PcdsGenerator struct {
+	Server      *DiscoveryServer
+	TrustBundle *tb.TrustBundle
+}
+
+var _ model.XdsResourceGenerator = &PcdsGenerator{}
+
+func pcdsNeedsPush(req *model.PushRequest) bool {
```

Comment:
I would just return 'false' for now - it is ok to have trust anchor update only on connect, it's not a frequent event.

Better than pushing too much ( i.e. on each config updated )

Reply:
One thing about this new code is it is unfortunately on by default. However, it only helps a small portion of users. So we need to optimize for the 99% of users that don't care, and not suddenly send a bunch of useless pushes.

So I agree with Costin, if we cannot make it push exactly when needed I would rather never push

Reply:
@myidpt - Correct me if I am wrong, but for CA migration, we WILL need changes in the meshConfig to be propagated to existing workloads. I don't think trust anchor update on connect will be sufficient. We need to find a way to ensure TrustAnchor configs ARE pushed as efficiently as possible

Reply:
I added a ProxyConfig config schema which is the **only** config update that the PcdsGenerator responds to. This means that the proxyConfig by default will not be pushed in 99 percent of the updates. When there is a change in the trustbundle (extremely rare), all configs will be pushed. I feel like this is a reasonable optimization?

Reply:
Ignore, didn't do this correctly. Figuring out how to do this the right way..

Reply:
Will be fixing this in the subsequent PR

Reply:
Will fix this in the subsequent PR